src/web/mjs/connectors/mangaz.mjs

```diff
@@ -0,0 +1,83 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class Mangaz extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'mangaz';
+        super.label = 'Manga Library Z (マンガ図書館Z)';
+        this.tags = ['manga', 'japanese'];
+        this.url = 'https://www.mangaz.com';
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        for(let page = 0, run = true; run; page++) {
+            let mangas = await this._getMangasFromPage(page);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        const request = new Request(new URL('/title/addpage_renewal?query=&page='+page, this.url), {
+            method:'GET',
+            headers: {
+                'X-Requested-With': 'XMLHttpRequest'
+            }
+        });
+        const data = await this.fetchDOM(request, 'h4 > a');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, request.url),
+                title: element.text.trim()
+            };
+        });
+    }
+
+    async _getMangaFromURI(uri) {
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'li.title');
+        const id = uri.pathname + uri.search;
+        const title = data[0].textContent.trim();
+        return new Manga(this, id, title);
+    }
+
+    async _getPages(chapter) {
+        const script = `
+            new Promise(async (resolve,reject) => {
+                let g = JCOMI.namespace("JCOMI.document")
```

Comment:
Use more expressive variable names when used in larger scopes, avoid single characters
Bad: `let a = b.c(d)`
Good: `let minified = text.remove(spaces)`

Reply:
i was just using the same ones as the site did to avoid confusion while writing it